src/wasm-lib/grackle/src/native_functions/sketch/stdlib_functions.rs

```diff
@@ -140,25 +150,124 @@ impl Callable for LineTo {
         &self,
         ctx: &mut crate::native_functions::Context<'_>,
         args: Vec<EpBinding>,
+    ) -> Result<EvalPlan, CompileError> {
+        LineBare::call(ctx, args, LineBareOptions { at: At::AbsoluteXY })
+    }
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(Eq, PartialEq))]
+pub struct Line;
+
+impl Callable for Line {
+    fn call(
+        &self,
+        ctx: &mut crate::native_functions::Context<'_>,
+        args: Vec<EpBinding>,
+    ) -> Result<EvalPlan, CompileError> {
+        LineBare::call(ctx, args, LineBareOptions { at: At::RelativeXY })
+    }
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(Eq, PartialEq))]
+pub struct XLineTo;
+
+impl Callable for XLineTo {
+    fn call(
+        &self,
+        ctx: &mut crate::native_functions::Context<'_>,
+        args: Vec<EpBinding>,
+    ) -> Result<EvalPlan, CompileError> {
+        LineBare::call(ctx, args, LineBareOptions { at: At::AbsoluteX })
+    }
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(Eq, PartialEq))]
+pub struct XLine;
+
+impl Callable for XLine {
+    fn call(
+        &self,
+        ctx: &mut crate::native_functions::Context<'_>,
+        args: Vec<EpBinding>,
+    ) -> Result<EvalPlan, CompileError> {
+        LineBare::call(ctx, args, LineBareOptions { at: At::RelativeX })
+    }
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(Eq, PartialEq))]
+pub struct YLineTo;
+
+impl Callable for YLineTo {
+    fn call(
+        &self,
+        ctx: &mut crate::native_functions::Context<'_>,
+        args: Vec<EpBinding>,
+    ) -> Result<EvalPlan, CompileError> {
+        LineBare::call(ctx, args, LineBareOptions { at: At::AbsoluteY })
+    }
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(Eq, PartialEq))]
+pub struct YLine;
+
+impl Callable for YLine {
+    fn call(
+        &self,
+        ctx: &mut crate::native_functions::Context<'_>,
+        args: Vec<EpBinding>,
+    ) -> Result<EvalPlan, CompileError> {
+        LineBare::call(ctx, args, LineBareOptions { at: At::RelativeY })
+    }
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(Eq, PartialEq))]
+/// Exposes all the possible arguments the `line` modeling command can take.
+/// Reduces code for the other line functions needed.
+/// We do not expose this to the developer since it does not align with
+/// the documentation (there is no "lineBare").
+pub struct LineBare;
+
+/// Used to configure the call to handle different line variants.
+pub struct LineBareOptions {
+    /// Where to start coordinates at, ex: At::RelativeXY.
+    at: At,
+}
+
+impl LineBare {
+    fn call(
+        ctx: &mut crate::native_functions::Context<'_>,
+        args: Vec<EpBinding>,
+        opts: LineBareOptions,
     ) -> Result<EvalPlan, CompileError> {
         let mut instructions = Vec::new();
-        let fn_name = "lineTo";
-        // Get both required params.
+        let fn_name = "lineBare";
```

Comment:
This gets shown to the user in KCL error messages -- suggest it should be a parameter instead, where each function sets its own name as the parameter.